ui/apps/platform/src/hooks/useURLSort.ts

```diff
@@ -1,33 +1,22 @@
-import { useRef, useState } from 'react';
-import useDeepCompareEffect from 'use-deep-compare-effect';
+import { useRef } from 'react';
+import findIndex from 'lodash/findIndex';
+import intersection from 'lodash/intersection';
+import isEqual from 'lodash/isEqual';
 
-import useURLParameter, { HistoryAction } from 'hooks/useURLParameter';
-import { SortAggregate, SortDirection, SortOption, ThProps } from 'types/table';
-import { ApiSortOption } from 'types/search';
-import { isParsedQs } from 'utils/queryStringUtils';
+import useURLParameter, { HistoryAction, QueryValue } from 'hooks/useURLParameter';
+import { SortAggregate, SortOption, ThProps, isSortOption } from 'types/table';
+import { ApiSortOption, ApiSortOptionSingle } from 'types/search';
 
 export type GetSortParams = (
-    field: string,
-    aggregateBy?: SortAggregate
+    columnName: string,
+    fieldOptions?: SortAggregate | { field: string; aggregateBy?: SortAggregate }[]
 ) => ThProps['sort'] | undefined;
 
-export type UseURLSortProps = {
-    sortFields: string[];
-    defaultSortOption: SortOption;
-    onSort?: (newSortOption: SortOption) => void;
-};
-
-export type UseURLSortResult = {
-    sortOption: ApiSortOption;
-    setSortOption: (newSortOption: SortOption, historyAction?: HistoryAction | undefined) => void;
-    getSortParams: GetSortParams;
-};
```

Comment:
The hook's input/output types were moved closer to the function for readability.